File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/history/HistoryFragment.java
package com.example.myapplication.ui.fragment.history;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.recyclerview.widget.LinearLayoutManager;

import com.example.myapplication.databinding.FragmentHistoryBinding;
import com.example.myapplication.model.Order;
import com.example.myapplication.util.android.base.BaseFragment;

import java.util.ArrayList;

public class HistoryFragment extends BaseFragment<FragmentHistoryBinding> {
    private ArrayList<Order> orders = new ArrayList<>();
    private HistoryAdapter adapter;

    private HistoryFragment() {}

    public static HistoryFragment newInstance() {
        return new HistoryFragment();
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        binding = FragmentHistoryBinding.inflate(inflater, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        requireActivity().setTitle("History");

        if (orders.size()==0){

        }

        adapter = new HistoryAdapter(orders);
        binding.recyclerviewHistory.setAdapter(adapter);
        binding.recyclerviewHistory.setHasFixedSize(true);
        binding.recyclerviewHistory.setLayoutManager(new LinearLayoutManager(getContext()));
    }
}

File: CustomerApp/app/src/main/java/com/example/myapplication/ui/fragment/history/HistoryFragment.kt
package com.example.myapplication.ui.fragment.history

import android.os.Bundle
import android.view.*
import androidx.fragment.app.viewModels
import androidx.lifecycle.observe
import androidx.navigation.fragment.findNavController
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.example.myapplication.CustomerApp
import com.example.myapplication.MobileNavigationDirections
import com.example.myapplication.R
import com.example.myapplication.databinding.FragmentHistoryBinding
import com.example.myapplication.db.CustomerDatabase
import com.example.myapplication.util.android.base.BaseFragment
import com.example.myapplication.util.android.restartApp
import com.example.myapplication.util.android.showIf
import org.kodein.di.KodeinAware
import org.kodein.di.android.x.kodein
import org.kodein.di.generic.instance

class HistoryFragment: BaseFragment<FragmentHistoryBinding>(), KodeinAware {
    override val kodein by kodein()
    val database by instance<CustomerDatabase>()
    val app by instance<CustomerApp>()

    private val historyViewModel by viewModels<HistoryViewModel> { HistoryViewModelFactory(database) }
    private val adapter by lazy(LazyThreadSafetyMode.NONE) { HistoryAdapter(mutableListOf(), database) }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setHasOptionsMenu(true)
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        _binding = FragmentHistoryBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.recyclerOrders.layoutManager = LinearLayoutManager(requireContext(), RecyclerView.VERTICAL, false)
        binding.recyclerOrders.adapter = adapter

        binding.recyclerOrders.showIf(adapter.itemCount != 0)
        binding.textEmptyState.showIf(adapter.itemCount == 0)

        historyViewModel.orders.observe(viewLifecycleOwner) {
            adapter.submitList(it)
            binding.recyclerOrders.showIf(adapter.itemCount != 0)
            binding.textEmptyState.showIf(adapter.itemCount == 0)
        }
    }

    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) =
        inflater.inflate(R.menu.menu_main, menu)

    override fun onOptionsItemSelected(item: MenuItem): Boolean =
        when(item.itemId) {
            R.id.action_settings -> {
                findNavController().navigate(MobileNavigationDirections.navigateToSettings())
                true
            }
            R.id.action_logout -> {
                app.signOut()
                restartApp(requireActivity())
                true
            }
            else -> super.onOptionsItemSelected(item)
        }
}